src/main/kotlin/com/github/hanseter/json/editor/ui/skins/ToggleSwitchSkin.kt

```diff
@@ -0,0 +1,233 @@
+        )
+    }
+
+    override fun layoutChildren(
+        x: Double, y: Double,
+        w: Double, h: Double
+    ) {
+        val checkBox = skinnable
+        val boxWidth = snapSize(box.prefWidth(-1.0))
+        val boxHeight = snapSize(box.prefHeight(-1.0))
+        val computeWidth = max(checkBox.prefWidth(-1.0), checkBox.minWidth(-1.0))
+        val labelWidth = min(computeWidth - boxWidth, w - snapSize(boxWidth))
+        val labelHeight = min(checkBox.prefHeight(labelWidth), h)
+        val maxHeight = max(boxHeight, labelHeight)
+        val xOffset = computeXOffset(w, labelWidth + boxWidth, checkBox.alignment.hpos) + x
+        val yOffset = computeYOffset(h, maxHeight, checkBox.alignment.vpos) + x
+        layoutLabelInArea(xOffset + boxWidth, yOffset, labelWidth, maxHeight, checkBox.alignment)
+
+        box.resize(boxWidth, boxHeight)
+
+        // Each time the layout is done, recompute the thumb "selected" position and apply it to the transition target.
+        val transitionDist = boxWidth / 4
+        transition.fromX = -transitionDist
+        transition.toX = transitionDist
+
+        thumb.translateX = 0.0
+
+        if (transition.status == Animation.Status.RUNNING) {
+            // If the transition is running, it must be restarted for the value to be properly updated.
+            val currentTime = transition.currentTime
+            transition.stop()
+            transition.playFrom(currentTime)
+        } else {
+            // If the transition is not running, simply apply the translate value.
+            thumb.translateX = when {
+                !checkBox.isSelected -> transition.fromX
+                checkBox.isSelected -> transition.toX
+                checkBox.isIndeterminate -> 0.0
+                else -> 0.0
+            }
+        }
+    }
+
+
+    private fun computeYOffset(h: Double, contentHeight: Double, vpos: VPos): Double {
+        return when (vpos) {
+            VPos.TOP -> 0.0
+            VPos.CENTER -> (h - contentHeight) / 2
+            VPos.BOTTOM -> h - contentHeight
+            else -> 0.0
+        }
+
+    }
+
+    private fun computeXOffset(w: Double, contentWidth: Double, hpos: HPos?): Double {
+        return when (hpos) {
+            HPos.LEFT -> 0.0
+            HPos.CENTER -> (w - contentWidth) / 2
+            HPos.RIGHT -> w - contentWidth
+            else -> 0.0
+        }
+    }
+}
+
+fun CheckBox.setNullableBoolean(newValue: Boolean?) {
+    when (newValue) {
+        true -> {
+            isSelected = true
+            isIndeterminate = false
+        }
+
+        false -> {
+            isSelected = false
+            isIndeterminate = false
+        }
+
+        null -> {
+            isIndeterminate = true
+            isSelected = false
+        }
+    }
+}
+
+fun CheckBox.getNullableBoolean(): Boolean? {
+    return when {
+        isSelected && !isIndeterminate -> true
+        !isSelected && !isIndeterminate -> false
+        else -> null
+    }
+}
```

Comment:
These extension method will be visible for everybody using this library. I think polluting everybody's auto-completion with only internally needed extension methods should be avoided. 
Please make these methods either internal or move them to `BooleanControl` and make them private. I would actually prefer the later.